cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/kafkaassigner/KafkaAssignerDiskUsageDistributionGoal.java

```diff
@@ -0,0 +1,568 @@
+   * 
+   * In addition, r1 and r2 must have the same role.
+   * @param r1 the first replica to swap
+   * @param r2 the second replica to swap with the first replica
+   * @param clusterModel the cluster model
+   * @return true if the two replicas can be swapped, false otherwise.
+   */
+  boolean canSwap(Replica r1, Replica r2, ClusterModel clusterModel) {
+    boolean inSameRack = r1.broker().rack() == r2.broker().rack() && r1.broker() != r2.broker();
+    boolean rackAware = !clusterModel.partition(r1.topicPartition()).partitionRacks().contains(r2.broker().rack()) 
+        && !clusterModel.partition(r2.topicPartition()).partitionRacks().contains(r1.broker().rack());
+    boolean sameRole = r1.isLeader() == r2.isLeader();
+    return (inSameRack || rackAware) && sameRole;
+  }
+
+  /**
+   * Find the position of the target size in the sorted replica list. The position is the index of the replica whose
+   * size is closest to but greater than (or less than) the target size.
+   * 
+   * @param sortedReplicas the replica list sorted in ascending order based on the size.
+   * @param targetSize the replica size to find.
+   * @param shiftOnExactMatch the position to shift if there is an exact match. The value should be 1 or 0 or -1.
+   *                          When the value is 1, the method returns the position of the first replica whose size is 
+   *                          greater than the target size.
+   *                          When the value is 0, the method returns the position of the first replica whose size is 
+   *                          greater than or equals to the target size.
+   *                          When the value is -1, the method returns the position of the replica whose size is 
+   *                          just less than the target size.
+   *                          
+   * @return the index of the replica whose size is closest to but greater than (or equals to if inclusive=true) the 
+   * target size
+   */
+  private int findReplicaPos(List<ReplicaWrapper> sortedReplicas, double targetSize, int shiftOnExactMatch) {
+    if (shiftOnExactMatch != 1 && shiftOnExactMatch != -1 && shiftOnExactMatch != 0) {
+      throw new IllegalArgumentException("The shift on exact match must be either 1 or -1");
+    }
+    int index = Collections.binarySearch(sortedReplicas, new ReplicaWrapper(null, targetSize),
+                                         Comparator.comparingDouble(ReplicaWrapper::size));
+    switch (shiftOnExactMatch) {
+      case -1:
+        return index >= 0 ? index - 1 : -(index + 1) - 1;
+      case 1:
+        return index >= 0 ? index + 1 : -(index + 1);
+      case 0:
+        return index >= 0 ? index : -(index + 1);
+      default:
+        throw new IllegalStateException("Invalid shift on exact match value " + shiftOnExactMatch);
+    }
+  }
+  
+  private List<Broker> sortedBrokersAboveSizeDescend(ClusterModel clusterModel, double aboveUsage) {
```

Comment:
Missing JavaDoc.

Reply:
This may vary but for most of the projects I have worked on, if a method is private and the logic is straight forward enough, a java doc is usually not mandatory.